### ISSUE TITLE
if challengeEndpoint is not defined, the solution endpoint defaults to microsoft API

### ISSUE THREAD

**Issue by lemoustachiste** [decentralized-identity/ion-tools#11]
As per https://github.com/decentralized-identity/ion-tools/blob/main/ion.js/lib.js#L96, one cannot specify only one of the endpoints otherwise they both are reset to the default microsoft endpoints.

Subsequently, why is the challenge endpoint not available locally? Or did I miss it?

**Comment by csuwildcat** [comment]
@lemoustachiste the challenge and the submission must be from the same ION node instance (because they are linked request/response endpoints that have data shared between them). For this reason, you can't take a challenge from one place and submit it somewhere else, because the server you are submitting to is the one that you need to satisfy the POW challenge for.

**Comment by lemoustachiste** [comment]
Thanks for your reply, I get the logic behind it.
However it seems I didn't have the challenge endpoint on my local node?

**Comment by csuwildcat** [comment]
@lemoustachiste yes, we didn't add the POW challenge setup to the node's code (it's just a thing we needed at Microsoft for ours) because we didn't think people would really be running open write nodes like we are (we also didn't have the time to generalize it as a contribution). Is this needed for your uses? We can think about landing that here, if there's a strong community desire.

**Comment by lemoustachiste** [comment]
@csuwildcat it's too early for me to know exactly what we'll need since I'm still at PoC stage. I think we have a catch up today so we can discuss that then. I guess it all boils down to if you want to maintain an endpoint for consumers to reach or, if consumers are going to run a node anyway, they might as be autonomous all the way through.
